### PR TITLE
Support rz_read_<endianness>_float/rz_read_<endianness>_double

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
Add a few unit tests please :)

**Review by ret2libc — COMMENTED** [review]
I know nothing about these... but yeah, just very simple unit tests for coverage would be great.
Moreover, for curiosity:
* is `me` Middle Endian?
* is this really how float/double numbers are stored on little/big endian machines?

**Comment by XVilka** [comment]
`me` is middle endian, yes.

File: test/unit/test_endian.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: 2019 ret2libc <sirmy15@gmail.com>
```
Comment (XVilka) [inline test/unit/test_endian.c]: Why not to put yourself here?

Comment (ret2libc) [reply test/unit/test_endian.c]: because he likes me <3

**Comment by XVilka** [comment]
Also clang-format.

**Comment by wargio** [comment]
> * is `me` Middle Endian?
> * is this really how float/double numbers are stored on little/big endian machines?

Yes, they always follow the IEEE-754 standard, and then is stored in memory like an ut32/ut64 based on the arch endianness.

**Review by XVilka — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**